steps/post_push/lib/docker-pull-resources/post-receive-hook

```diff
@@ -16,4 +16,10 @@ done
 
 cd $DEPLOY_DIR/current && docker-compose -f docker-compose.yml up -d --no-build
 
+if [ ! -z "$LOG_DIR" ]; then
```

Comment:
Is `-z` usable in a POSIX (non-bash) shell? I haven't seen it outside of bash before so need confirmation.

Also this is checking if value for `$LOG_DIR` is set, do we also need a check for the directory existing (which should also be a valid check in case the path is a symlink to another directory)?

Reply:
Hi,
`-z` seems usable in POSIX shell. As it takes `-z` as an operand for `test` command and I understood we can use `if test <condition>` and `if [ <condition> ]` interchangeably.

but since I am using ` ! -z ` i saw it's recommended to use `-n` operand instead, so I've changed it to that. 
 
I also added another check for `$LOG_DIR` so that it will only try to create the folder if `$LOG_DIR` folder doesn't exist.

- [`test` command documentation ](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/test.html)
- [`test <condition>` command and `if [ <condition> ]` can be used interchangeably ](https://wiki.bash-hackers.org/commands/classictest)